test/functional/p2p_segwit.py

```diff
@@ -1956,23 +1956,38 @@ def test_non_standard_witness(self):
     def test_upgrade_after_activation(self):
         """Test the behavior of starting up a segwit-aware node after the softfork has activated."""
 
-        self.restart_node(2, extra_args=["-segwitheight={}".format(SEGWIT_HEIGHT)])
+        # All nodes are caught up and node 2 is a pre-segwit node that will soon upgrade.
+        assert_equal(self.nodes[0].getblockcount(), self.nodes[2].getblockcount())
+        assert_equal(self.nodes[1].getblockcount(), self.nodes[2].getblockcount())
+        assert SEGWIT_HEIGHT < self.nodes[2].getblockcount()
+        assert softfork_active(self.nodes[0], 'segwit')
+        assert softfork_active(self.nodes[1], 'segwit')
+        assert 'segwit' not in self.nodes[2].getblockchaininfo()['softforks']
+
+        # Restarting node 2 should result in a shutdown because the blockchain consists of
+        # insufficiently validated blocks per segwit consensus rules.
+        self.stop_node(2)
+        with self.nodes[2].assert_debug_log(expected_msgs=[
+                'Segwit blocks are insufficiently validated. Please restart with -reindex.'], timeout=10):
+            self.nodes[2].start(["-segwitheight={}".format(SEGWIT_HEIGHT)])
```

Comment:
(please don't change the branch for this, but if you need to retouch for some other reason) you can use an f-string here:

```suggestion
            self.nodes[2].start([f"-segwitheight={SEGWIT_HEIGHT}"])
```

f-strings are supported from python 3.6, and have a slightly cleaner syntax.

You can also use an f-string in the `extra_args` argument below.

Reply:
Thanks for showing me that. Done.